plugin-server/functional_tests/ingestion.test.ts

```diff
@@ -317,108 +272,197 @@ test.concurrent('consumer updates timestamp exported to prometheus', async () =>
+    const initialEventId = new UUIDT().toString()
+    await capture(producer, teamId, initialDistinctId, initialEventId, 'custom event')
+    const secondEventId = new UUIDT().toString()
+    await capture(producer, teamId, secondDistinctId, secondEventId, 'custom event 2')
+    await waitForExpect(async () => {
+        const persons = await fetchPersons(clickHouseClient, teamId)
+        expect(persons).toEqual(
+            expect.arrayContaining([
                 expect.objectContaining({
                     properties: expect.objectContaining({ $creator_event_uuid: initialEventId }),
-                })
-            )
-        }, 10000)
-
-        // We then have the user identify themselves, but on e.g. a different
-        // device and hence a different anonymous id.
-        const initialIdentifyEventId = new UUIDT().toString()
-        await capture(producer, teamId, personIdentifier, initialIdentifyEventId, '$identify', {
-            $anon_distinct_id: secondDistinctId,
-            distinct_id: personIdentifier,
-        })
-        await waitForExpect(async () => {
-            const persons = await fetchPersons(clickHouseClient, teamId)
-            expect(persons).toContainEqual(
+                }),
                 expect.objectContaining({
-                    properties: expect.objectContaining({ $creator_event_uuid: initialIdentifyEventId }),
-                })
-            )
-        }, 10000)
-
-        // Then we create another event with the initial anonymous distinct id,
-        // shortly followed by another identify event but this time with the
-        // initial anonymous distinct id
-        const uuidOfEventThatShouldBeIdentified = new UUIDT().toString()
-        await capture(producer, teamId, initialDistinctId, uuidOfEventThatShouldBeIdentified, 'custom event')
-
-        const uuidOfIdentifyEvent = new UUIDT().toString()
-        await capture(producer, teamId, personIdentifier, uuidOfIdentifyEvent, '$identify', {
-            distinct_id: personIdentifier,
-            $anon_distinct_id: initialDistinctId,
-        })
+                    properties: expect.objectContaining({ $creator_event_uuid: secondEventId }),
+                }),
+            ])
+        )
+    }, 10000)
+
+    // Then we identify both ids
+    const uuidOfFirstIdentifyEvent = new UUIDT().toString()
+    await capture(producer, teamId, personIdentifier, uuidOfFirstIdentifyEvent, '$identify', {
+        distinct_id: personIdentifier,
+        $anon_distinct_id: initialDistinctId,
+    })
+    const uuidOfSecondIdentifyEvent = new UUIDT().toString()
+    await capture(producer, teamId, personIdentifier, uuidOfSecondIdentifyEvent, '$identify', {
+        distinct_id: personIdentifier,
+        $anon_distinct_id: secondEventId,
+    })
+
+    let events: ClickHouseEvent[] = []
+    await waitForExpect(async () => {
+        events = await fetchEvents(clickHouseClient, teamId)
+        expect(events.length).toBe(4)
+    }, 10000)
```

Comment:
```suggestion
    const events = await waitForExpect(async () => {
        const events = await fetchEvents(clickHouseClient, teamId)
        expect(events.length).toBe(4)
        return events
    }, 10000)
```

It's a tossup but this way removes any setting of variables in parent scopes which I find confusing.